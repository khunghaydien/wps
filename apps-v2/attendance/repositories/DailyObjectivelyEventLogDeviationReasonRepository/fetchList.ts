import Api from '@apps/commons/api';

import { IDailyObjectivelyEventLogDeviationReasonRepository } from '@attendance/domain/models/DailyObjectivelyEventLogDeviationReason';

export type Response = {
  id: string;
  deviationReasons: {
    label: string;
    value: string;
  }[];
};

export default (async ({ employeeId, targetDate }) => {
  const response: Response = await Api.invoke({
    path: '/att/objectively-event-log/deviated-reason/list',
    param: {
      empId: employeeId,
      targetDate,
    },
  });
  return {
    employeeId,
    targetDate,
    id: response.id,
    deviationReasons: new Map(
      response.deviationReasons?.map((value) => [value.value, value]) || []
    ),
  };
}) as IDailyObjectivelyEventLogDeviationReasonRepository['fetchList'];
