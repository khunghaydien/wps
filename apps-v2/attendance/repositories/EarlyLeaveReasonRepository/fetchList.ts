import Api from '../../../commons/api';

import { IEarlyLeaveReasonRepository } from '@attendance/domain/models/EarlyLeaveReason';

type Response = Readonly<{
  earlyLeaveReasons: {
    id: string; // 早退理由id
    code: string; // 早退理由コード
    name: string; // 早退理由名
    earlyLeaveEndTime: number | null; // 終了時間
  }[];
}>;

const fetchList: IEarlyLeaveReasonRepository['fetchList'] = async ({
  targetDate,
  employeeId,
}) => {
  const response: Response = await Api.invoke({
    path: '/att/daily-early-leave-reason/list',
    param: { targetDate, empId: employeeId },
  });
  return response.earlyLeaveReasons;
};

export default fetchList;
