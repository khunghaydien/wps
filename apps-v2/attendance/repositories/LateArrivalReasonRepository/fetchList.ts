import Api from '../../../commons/api';

import {
  ILateArrivalReasonRepository,
  LateArrivalReason as DomainLateArrivalReason,
} from '@attendance/domain/models/LateArrivalReason';

type Response = Readonly<{
  lateArrivalReasons: DomainLateArrivalReason[];
}>;

const fetchList: ILateArrivalReasonRepository['fetchList'] = async ({
  targetDate,
  employeeId,
}) => {
  const response: Response = await Api.invoke({
    path: '/att/daily-late-arrival-reason/list',
    param: { targetDate, empId: employeeId },
  });
  return response.lateArrivalReasons;
};

export default fetchList;
