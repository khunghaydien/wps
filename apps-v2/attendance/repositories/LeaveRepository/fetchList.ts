import Api from '@apps/commons/api';

import * as DomainLeave from '@attendance/domain/models/Leave';
import * as DomainLeaveRange from '@attendance/domain/models/LeaveRange';
import * as DomainLeaveType from '@attendance/domain/models/LeaveType';

import createMapByCode from '@attendance/libraries/utils/Array/createMapByCode';

export type Leave = {
  name: string;
  code: string;
  type: DomainLeaveType.LeaveType;
  ranges: DomainLeaveRange.LeaveRange[];
  details: DomainLeave.Detail[];
  daysLeft: number | null;
  hoursLeft: number | null;
  timeLeaveDaysLeft: number | null;
  timeLeaveHoursLeft: number | null;
  requireReason: boolean;
};

export type Response = {
  leaves: Leave[];
};

const convert = (param: Leave): DomainLeave.Leave => ({
  ...DomainLeave.defaultValue,
  ...param,
  ranges: DomainLeaveRange.sort(param.ranges) || [],
  details: createMapByCode<DomainLeave.Detail>(
    param.details?.map((value) => ({
      ...value,
      ranges: DomainLeaveRange.sort(value.ranges) || [],
    }))
  ),
});

export default (async (param) =>
  Api.invoke({
    path: '/att/daily-leave/list',
    param: {
      targetDate: param.targetDate || null,
      empId: param.employeeId || null,
      ignoredId: param.ignoredId || null,
    },
  }).then(({ leaves }: Response) =>
    createMapByCode(leaves.map((value) => convert(value)))
  )) as DomainLeave.ILeaveRepository['fetchList'];
