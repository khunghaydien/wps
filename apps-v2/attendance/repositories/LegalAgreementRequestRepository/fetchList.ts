import Api from '@apps/commons/api';
import TimeUtil from '@apps/commons/utils/TimeUtil';

import {
  ILegalAgreementRequestRepository,
  LegalAgreementRequestList,
} from '../../domain/models/LegalAgreementRequest';

const fetchList: ILegalAgreementRequestRepository['fetchList'] = async ({
  employeeId,
  targetDate,
}): Promise<LegalAgreementRequestList> => {
  const result: LegalAgreementRequestList = await Api.invoke({
    path: '/att/request/legal-agreement/list',
    param: {
      empId: employeeId,
      targetDate,
    },
  });
  return {
    ...result,
    requests: result.requests?.map((request) => ({
      ...request,
      changedOvertimeHoursLimit: TimeUtil.toHours(
        request?.changedOvertimeHoursLimit
      ),
    })),
  };
};
export default fetchList;
