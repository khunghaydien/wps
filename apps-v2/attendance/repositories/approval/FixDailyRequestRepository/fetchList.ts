import Api from '@apps/commons/api';

import {
  AttendanceSummary,
  createDividedSummaries,
  createOwnerInfos,
  createRecord,
  createSummaries,
  createWorkingType,
} from '@attendance/repositories/approval/models/AttendanceSummary';

import {
  FixDailyRequest,
  IFixDailyRequestRepository,
  Status,
} from '@attendance/domain/models/approval/FixDailyRequest';

export type Response = {
  requestList: {
    id: string;
    status: Status;
    employeeInfoList: {
      startDate: string;
      endDate: string;
      departmentName: string;
      workingTypeName: string;
    }[];
    employeeCode: string;
    employeeName: string;
    photoUrl: string;
    departmentName: string;
    delegatedEmployeeName: string;
    // comment: string;
    requestDate: string;
    targetDate: string;
    record: AttendanceSummary['records'][number];
    recordTotal: {
      restTime: number;
      realWorkTime: number;
      overTime: number;
      nightTime: number;
      lostTime: number;
      virtualWorkTime: number;
      holidayWorkTime: number;
    };
    useRestReason: boolean;
    summaries: AttendanceSummary['summaries'];
    dividedSummaries: AttendanceSummary['dividedSummaries'];
  }[];
};

const $convert = (response: Response): FixDailyRequest[] => {
  const { requestList } = response;
  return requestList.map(
    ({
      photoUrl: employeePhotoUrl,
      departmentName,
      delegatedEmployeeName,
      record,
      recordTotal,
      useRestReason,
      ...request
    }) => {
      const targetRecord = createRecord(record);
      return {
        id: request.id,
        requestDate: request.requestDate,
        targetDate: request.targetDate,
        startDate: null,
        endDate: null,
        submitter: {
          employee: {
            name: request.employeeName,
            code: request.employeeCode,
            photoUrl: employeePhotoUrl,
            department: {
              name: departmentName,
            },
          },
          delegator: {
            employee: {
              name: delegatedEmployeeName,
            },
          },
        },
        ownerInfos: createOwnerInfos(request),
        targetRecord,
        records: [targetRecord],
        recordTotal,
        useRestReason: false,
        summaries: createSummaries(request.summaries, { records: [record] }),
        dividedSummaries: createDividedSummaries(request.dividedSummaries, {
          records: [record],
        }),
        attention: {
          ineffectiveWorkingTime: 0,
          insufficientRestTime: 0,
        },
        workingType: createWorkingType({ records: [record], useRestReason }),
        dailyAllowanceRecords: null,
        dailyObjectiveEventLogRecords: null,
        dailyRestRecords: null,
        displayFieldLayout: null,
      };
    }
  );
};

const fetchList: IFixDailyRequestRepository['fetchList'] = async ({
  employeeId,
  approvalType,
}) => {
  const response: Response = await Api.invoke({
    path: '/att/request-list/fix-daily/get',
    param: {
      empId: employeeId,
      approvalType,
    },
  });
  return $convert(response);
};

export default fetchList;
