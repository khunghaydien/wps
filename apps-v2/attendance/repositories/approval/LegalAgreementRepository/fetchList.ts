import Api from '@apps/commons/api';

import {
  ILegalAgreementRequestRepository,
  Status,
} from '@apps/attendance/domain/models/approval/LegalAgreementRequest';
import { Code } from '@attendance/domain/models/LegalAgreementRequestType';

type Response = {
  requestList: {
    id: string;
    employeeName: string;
    delegatedEmployeeName: string | null;
    photoUrl: string;
    departmentName: string | null;
    approverName: string;
    approverPhotoUrl: string;
    approverDepartmentName: string | null;
    requestDate: string;
    targetMonth: string;
    requestType: Code;
    requestStatus: Status;
    originalRequestStatus: Status;
  }[];
};

const fetchList: ILegalAgreementRequestRepository['fetchList'] = async (
  approvalType,
  selectId
) => {
  const response: Response = await Api.invoke({
    path: '/att/request-list/legal-agreement/get',
    param: { approvalType, selectId },
  });
  return response.requestList;
};

export default fetchList;
