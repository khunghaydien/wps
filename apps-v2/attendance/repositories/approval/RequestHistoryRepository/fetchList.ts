import Api from '@commons/api';

import adapter from '@apps/repositories/adapters';

import { IRequestHistoryRepository } from '@attendance/domain/models/approval/RequestHistory';

export default ((requestId) => {
  return Api.invoke({
    path: '/approval/request/history/get',
    param: adapter.toRemote({
      requestId,
    }),
  }).then(({ historyList }) =>
    adapter.fromRemote(historyList.map((history) => ({ ...history })))
  );
}) as IRequestHistoryRepository['fetchList'];
