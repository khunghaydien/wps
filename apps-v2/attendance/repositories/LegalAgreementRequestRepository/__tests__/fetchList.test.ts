import Api from '../../../../../__tests__/mocks/ApiMock';
import fetchList from '../fetchList';
import { defaultValue } from './mocks/fetchList.mock';

it('should return legalAgreement request list.', async () => {
  // Arrange
  const MockApi = Api.invoke as jest.Mock;
  MockApi.mockResolvedValueOnce(defaultValue);

  // Act
  const list = await fetchList({
    employeeId: 'employeeId',
    targetDate: '2022-09-23',
  });

  // Assert
  expect(list).toMatchSnapshot();
  expect(MockApi).toBeCalledTimes(1);
  expect(MockApi).toBeCalledWith({
    path: '/att/request/legal-agreement/list',
    param: {
      empId: 'employeeId',
      targetDate: '2022-09-23',
    },
  });
});
