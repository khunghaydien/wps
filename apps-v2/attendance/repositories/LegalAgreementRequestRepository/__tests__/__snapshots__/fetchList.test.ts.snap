// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return legalAgreement request list. 1`] = `
Object {
  "availableRequestTypes": Array [
    "Monthly",
  ],
  "requests": Array [],
}
`;
